scipy/optimize/_minimize.py

```diff
@@ -321,6 +321,9 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
     It should converge to the theoretical solution (1.4 ,1.7).
 
     """
+    x0 = np.asarray(x0)
+    if x0.dtype.kind in np.typecodes["AllInteger"]:
+        x0 = np.asarray(x0, dtype=float)
```

Comment:
that's indeed shorter here, will change. Normally I prefer the explicit ``dtype=`` for readability, but here it's 3 lines.

``asfarray`` needs some TLC it looks like: ``def asfarray(a, dtype=_nx.float_)`` :(